client/container.go

```diff
@@ -186,8 +186,17 @@ func (c *container) Delete(ctx context.Context, opts ...DeleteOpts) error {
 		tracing.WithAttribute("container.id", c.id),
 	)
 	defer span.End()
-	if _, err := c.loadTask(ctx, nil); err == nil {
-		return fmt.Errorf("cannot delete running task %v: %w", c.id, errdefs.ErrFailedPrecondition)
+	ts, err := c.loadTask(ctx, nil)
```

Comment:
what about this change?

```go
ct, err := c.loadTask(ctx, nil)
if err == nil {
     ts, err := ct.Status(ctx)
     if err == nil && ts.Status == Running {
              return fmt.Errorf("cannot delete running task %v: %w", c.id, errdefs.ErrFailedPrecondition)
     }
}
```

If it's not running, let's fall through to original cleanup.

Reply:
If I don't send a signal to Shim here, containerd will delete this record, but Shim doesn't know, which will result in Shim residue.

Reply:
OK. After revisit code, I think we should not change the logic here, since the task could be in paused state. `container.Delete` should block here if task record is still in shim side. The caller should clean up task first before deleting task.

Reply:
Sure, it should call shim to delete the error container in caller, I will change it.